crates/db-allocator/src/lib.rs

```diff
@@ -48,35 +70,82 @@ impl Constraint {
     }
 
     /// Set the min constraint.
-    pub fn min<T>(mut self, min: T) -> Self
+    pub fn min<T>(&mut self, min: T)
```

Comment:
The accessors should return "&mut self", otherwise it can't support tail call as:
```
constraint.min(10).max(100).align(8)
```

Reply:
fixed~